### PR TITLE
Add optional fuzzy search algorithm

### PR DESCRIPTION
**Pull request by jorismertz**
I discovered that the current search algorithm struggles with queries that start with a single letter followed by a space. To address this issue, I've added a new feature flag that enables fuzzy searching. This makes searching a lot more flexible and also sorts the results by their relevance.

This is my first time using feature flags, so I am open to feedback on the control flow implementation.

### CODE REVIEW

File: spotify_player/src/state/ui/mod.rs [hunk 1]
```diff
@@ -83,6 +83,11 @@ impl UIState {
         match self.popup {
             Some(PopupState::Search { ref query }) => {
                 let query = query.to_lowercase();
+
+                #[cfg(feature = "fzf")]
+                return fuzzy_search_items(items, &query);
```
Comment (aome510) [inline spotify_player/src/state/ui/mod.rs:88]:
```suggestion
               fuzzy_search_items(items, &query)
```

`return` is redundant here

File: spotify_player/src/state/ui/mod.rs [hunk 2]
```diff
@@ -100,6 +105,27 @@ impl UIState {
...
+            matcher
+                .fuzzy(&t.to_string(), &query, false)
+                .map(|(score, _)| (t, score))
+        })
+        .collect::<Vec<_>>();
+
+    result.sort_by(|(_, a), (_, b)| b.cmp(a));
+    let result = result.into_iter().map(|(t, _)| t).collect::<Vec<_>>();
+
+    return result;
```
Comment (aome510) [inline spotify_player/src/state/ui/mod.rs:126]:
```suggestion
    result
```

same here

**Review by aome510 — APPROVED** [review]
thanks, LGTM!

File: spotify_player/src/state/ui/mod.rs (hunk 1 above)
Comment (jorismertz) [reply spotify_player/src/state/ui/mod.rs:88]: I thought this too but it won't let me compile.
```
error: expected `;`, found `#`
  --> spotify_player/src/state/ui/mod.rs:88:50
   |
88 |                 fuzzy_search_items(items, &query)
   |                                                  ^ help: add `;` here
89 |
90 |                 #[cfg(not(feature = "fzf"))]
   |                 - unexpected token
```

**Comment by jorismertz** [comment]
Accidentally included a commit i did not mean to.

File: spotify_player/src/state/ui/mod.rs (hunk 1 above)
Comment (aome510) [reply spotify_player/src/state/ui/mod.rs:88]: right, I forgot

**Comment by aome510** [comment]
Look like CI failed but it's unrelated to your change. I can take a look and fix later today

**Pull request merged**